### PR TITLE
Test HF's implementation of  Qwen 2 model

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds a test to verify that the computed cross-entropy loss matches between HF's implementation and application of Thunder on this model.

Currently backward doesn't work because it hits a bug in nvFuser (https://github.com/NVIDIA/Fuser/issues/871#issuecomment-2461562917).

Ref https://github.com/Lightning-AI/lightning-thunder/issues/1286.

### CODE REVIEW

**Comment by IvanYashchuk** [comment]
Interesting:
```
        # https://github.com/NVIDIA/Fuser/issues/871#issuecomment-2461562917
>       with pytest.raises(RuntimeError, match="Found two different const extents in the same set"):
E       Failed: DID NOT RAISE <class 'RuntimeError'>

thunder/tests/test_networks.py:443: Failed
```

File: thunder/tests/test_networks.py [hunk 1]
```diff
@@ -400,6 +400,47 @@ def test_thunderfx_mistral_nemo_small():
...
+    compiled_output = compiled_model(input_ids=input_ids, labels=input_ids)
+    compiled_loss = compiled_output.loss
+
+    # Less strict tolerance probably due to different type promotion order for bfloat16
+    # TODO: Investigate why the loss is different
+    # https://github.com/Lightning-AI/lightning-thunder/issues/1407
+    torch.testing.assert_close(compiled_loss, ref_loss, rtol=1e-4, atol=1e-4)
+
+    assert len(backend.subgraph_infos) == 1, "Should have exactly 1 subgraph because of fullgraph=True"
+    compiled_loss.backward(torch.randn_like(compiled_loss))
```
[the remark this answers is not in the document]
Comment (IvanYashchuk) [reply thunder/tests/test_networks.py:441]: Yes, let's add it. It didn't work initially with my nvFuser installation, but grad comparison should be added still.

**Comment by riccardofelluga** [comment]
It seems like the backward issue is not there anymore. As of time of writing, both CI and the latest container do not seem to present the issue.

**Review by t-vi — APPROVED** [review]
Thank you @IvanYashchuk  @crcrpar @riccardofelluga
